### PR TITLE
TP: converts DS tables to more performant and powerful ag-grid

### PR DESCRIPTION
**Pull request by mitchell852**
## What does this PR (Pull Request) do?

- [x] This PR is related to #4354

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?

- Run UI tests
- Navigate to one of the delivery service tables and interact with it
- 
https://tp.domain.tld/#!/delivery-services
https://tp.domain.tld/#!/cdns/{{cdn-id}}/delivery-services
https://tp.domain.tld/#!/profiles/{{profile-id}}/delivery-services
https://tp.domain.tld/#!/service-categories/{{service-category-name}}/delivery-services
https://tp.domain.tld/#!/tenants/{{tenant-id}}/delivery-services
https://tp.domain.tld/#!/topologies/delivery-services?name={{topology-name}}
https://tp.domain.tld/#!/types/{{type-id}}/delivery-services
https://tp.domain.tld/#!/servers/{{server-id}}/delivery-services

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js [hunk 1]
```diff
@@ -17,30 +17,385 @@
...
+        },
+        {
+            headerName: "Max DNS Answers",
+            field: "maxDnsAnswers",
+            hide: true
+        },
+        {
+            headerName: "Max Origin Connections",
+            field: "maxOriginConnections",
+            hide: true
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js:251]: same as above RE: filter

File: traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js [hunk 2]
```diff
@@ -17,30 +17,385 @@
...
+        },
+        {
+            headerName: "Long Desc 3",
+            field: "longDesc2",
+            hide: true
+        },
+        {
+            headerName: "Max DNS Answers",
+            field: "maxDnsAnswers",
+            hide: true
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js:246]: same as above RE: filter

File: traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js [hunk 3]
```diff
@@ -17,30 +17,385 @@
...
+        },
+        {
+            headerName: "Max Origin Connections",
+            field: "maxOriginConnections",
+            hide: true
+        },
+        {
+            headerName: "Max Request Header Bytes",
+            field: "maxRequestHeaderBytes",
+            hide: true
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js:256]: same as above RE: filter

File: traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js [hunk 4]
```diff
@@ -17,30 +17,385 @@
...
+        },
+        {
+            headerName: "Signing Algorithm",
+            field: "signingAlgorithm",
+            hide: true
+        },
+        {
+            headerName: "Range Slice Block Size",
+            field: "rangeSliceBlockSize",
+            hide: true
```
Comment (ocket8888) [inline traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js:354]: same as above RE: filter

**Review by ocket8888 — COMMENTED** [review]
Since there's a lot of duplication I just specified things in the first file in which I saw them.

File: traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js [hunk 5]
```diff
@@ -17,30 +17,385 @@
...
+        },
+        {
+            headerName: "DNS Bypass IPv6",
+            field: "dnsBypassIp6",
+            hide: true
+        },
+        {
+            headerName: "DNS Bypass TTL",
+            field: "dnsBypassTtl",
+            hide: true
```
[the remark this answers is not in the document]
Comment (mitchell852) [reply traffic_portal/app/src/common/modules/table/deliveryServices/TableDeliveryServicesController.js]: i did it for all of them in TP

File: traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html [hunk 6]
```diff
@@ -59,7 +59,7 @@
                     <li role="menuitem"><a ng-click="viewOrigins()">Manage Origins</a></li>
                     <li role="menuitem"><a ng-click="viewRegexes()">Manage Regexes</a></li>
                     <li role="menuitem"><a ng-click="viewCapabilities()">Manage Required Server Capabilities</a></li>
-                    <li role="menuitem"><a ng-click="viewServers()">Manage Servers</a></li>
+                    <li name="manageServersMenuItem" role="menuitem"><a ng-click="viewServers()">Manage Servers</a></li>
```
[the remark this answers is not in the document]
Comment (mitchell852) [reply traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html:62]: this is a pretty global issue in TP. i can change this one but i think maybe an issue to fix them all might be in order.

[the remark this answers is not in the document]
Comment (mitchell852) [reply traffic_portal/app/src/common/modules/form/deliveryService/form.deliveryService.DNS.tpl.html]: updated

**Review by ocket8888 — APPROVED** [review]
Seems to work well

**Pull request merged**